### PR TITLE
Issues #40-#46

### LINKED ISSUE #40: Laisser la possibilité de lire des dataframe au lieu de fichiers
Aujourd'hui csv-detective n'analyse qu'à partir d'un fichier. Laisser la possibilité d'analyser un dataframe également.

### LINKED ISSUE #42: FutureWarning when analysing CSV
When calling `routine` on https://static.data.gouv.fr/resources/adresses-au-format-bal-lacarre/20200319-151130/20200319
-bal-216402974.csv

```
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
/Users/alexandre/Developer/Etalab/udata-hydra/.venv/lib/python3.9/site-packages/csv_detective/utils.py:64: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  return_table.loc[key] = table.apply(lambda serie: test_col_val(
```

### LINKED ISSUE #43: date or datetime as python_type?
Would it make sense to output `date` or `datetime` as `python_type` for `date_der_maj` in the following example?

```json
{
   "encoding":"UTF-8-SIG",
   "separator":";",
   "header_row_idx":0,
   "header":[
      "cle_interop",
      "uid_adresse",
      "voie_nom",
      "numero",
      "suffixe",
      "commune_nom",
      "position",
      "x",
      "y",
      "long",
      "lat",
      "source",
      "date_der_maj",
      "refparc",
      "voie_nom_eu",
      "complement"
   ],
   "total_lines":82,
   "heading_columns":0,
   "trailing_columns":0,
   "continuous":[
      "x",
      "y",
      "long",
      "lat"
   ],
   "categorical":[
      "uid_adresse",
      "suffixe",
      "commune_nom",
      "position",
      "source",
      "complement"
   ],
   "columns_fields":{
      "cle_interop":{
         "python_type":"float",
         "format":"float",
         "score":1.0
      },
      "uid_adresse":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "voie_nom":{
         "python_type":"string",
         "format":"adresse",
         "score":1.0
      },
      "numero":{
         "python_type":"int",
         "format":"int",
         "score":1.0
      },
      "suffixe":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "commune_nom":{
         "python_type":"string",
         "format":"commune",
         "score":1.0
      },
      "position":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "x":{
         "python_type":"float",
         "format":"longitude_l93",
         "score":0.9795918367346939
      },
      "y":{
         "python_type":"float",
         "format":"latitude_l93",
         "score":1.0
      },
      "long":{
         "python_type":"float",
         "format":"latitude_wgs",
         "score":1.0
      },
      "lat":{
         "python_type":"float",
         "format":"longitude_wgs",
         "score":1.0
      },
      "source":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "date_der_maj":{
         "python_type":"string",
         "format":"date",
         "score":1.0
      },
      "refparc":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "voie_nom_eu":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "complement":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      }
   },
   "columns_labels":{
      "cle_interop":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "uid_adresse":{
         "python_type":"string",
         "format":"adresse",
         "score":0.5
      },
      "voie_nom":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "numero":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "suffixe":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "commune_nom":{
         "python_type":"string",
         "format":"commune",
         "score":0.5
      },
      "position":{
         "python_type":"string",
         "format":"latlon_wgs",
         "score":1.0
      },
      "x":{
         "python_type":"float",
         "format":"longitude_wgs_fr_metropole",
         "score":1.0
      },
      "y":{
         "python_type":"float",
         "format":"latitude_wgs_fr_metropole",
         "score":1.0
      },
      "long":{
         "python_type":"float",
         "format":"longitude_wgs_fr_metropole",
         "score":1.0
      },
      "lat":{
         "python_type":"float",
         "format":"latitude_wgs_fr_metropole",
         "score":1.0
      },
      "source":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "date_der_maj":{
         "python_type":"string",
         "format":"date",
         "score":1.0
      },
      "refparc":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "voie_nom_eu":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "complement":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      }
   },
   "columns":{
      "cle_interop":{
         "python_type":"float",
         "format":"float",
         "score":1.0
      },
      "uid_adresse":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "voie_nom":{
         "python_type":"string",
         "format":"adresse",
         "score":1.0
      },
      "numero":{
         "python_type":"int",
         "format":"int",
         "score":1.0
      },
      "suffixe":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "commune_nom":{
         "python_type":"string",
         "format":"commune",
         "score":1.25
      },
      "position":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "x":{
         "python_type":"float",
         "format":"longitude_l93",
         "score":1.4693877551020407
      },
      "y":{
         "python_type":"float",
         "format":"latitude_l93",
         "score":1.5
      },
      "long":{
         "python_type":"float",
         "format":"longitude_wgs_fr_metropole",
         "score":1.5
      },
      "lat":{
         "python_type":"float",
         "format":"latitude_wgs_fr_metropole",
         "score":1.5
      },
      "source":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "date_der_maj":{
         "python_type":"string",
         "format":"date",
         "score":1.5
      },
      "refparc":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "voie_nom_eu":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      },
      "complement":{
         "python_type":"string",
         "format":"string",
         "score":1.0
      }
   },
   "formats":{
      "float":[
         "cle_interop"
      ],
      "string":[
         "uid_adresse",
         "suffixe",
         "position",
         "source",
         "refparc",
         "voie_nom_eu",
         "complement"
      ],
      "adresse":[
         "voie_nom"
      ],
      "int":[
         "numero"
      ],
      "commune":[
         "commune_nom"
      ],
      "longitude_l93":[
         "x"
      ],
      "latitude_l93":[
         "y"
      ],
      "longitude_wgs_fr_metropole":[
         "long"
      ],
      "latitude_wgs_fr_metropole":[
         "lat"
      ],
      "date":[
         "date_der_maj"
      ]
   }
}
```

### LINKED ISSUE #44: Wrong type and header line detection
- header_row_idx should be 1 (there are two duplicate header lines)
- `NUMCOM` and `NUMDEP` should not be detected as int (Corsica forever)

http://data.caf.fr/dataset/f6411f07-10bf-4f13-b4fb-8d30ba9328b5/resource/94a182c4-19c8-4d3a-987c-187a49756365/download/txcouvglo2014.csv

```
[:~] $ head /Users/alexandre/Downloads/txcouvglo2014.csv
NUMCOM;NOMCOM;NUMDEP;NOMDEP;NUMEPCI;NOMEPCI;TXCOUVGLO_COM_2014;TXCOUVGLO_DEP_2014;TXCOUVGLO_EPCI_2014
NUMCOM;NOMCOM;NUMDEP;NOMDEP;NUMEPCI;NOMEPCI;TXCOUVGLO_COM_2014;TXCOUVGLO_DEP_2014;TXCOUVGLO_EPCI_2014
01001;L'ABERGEMENT-CLEMENCIAT;01;AIN;200035210;CC CHALARONNE CENTRE;41.7;65.2;72.9
01002;L'ABERGEMENT-DE-VAREY;01;AIN;240100883;CC DE LA PLAINE DE L'AIN;34.1;65.2;75.2
01004;AMBERIEU-EN-BUGEY;01;AIN;240100883;CC DE LA PLAINE DE L'AIN;61.8;65.2;75.2
01005;AMBERIEUX-EN-DOMBES;01;AIN;200042497;CC DOMBES SAONE VALLEE;73.6;65.2;77.8
01006;AMBLEON;01;AIN;200040350;CC BUGEY SUD;93.1;65.2;52.4
01007;AMBRONAY;01;AIN;240100883;CC DE LA PLAINE DE L'AIN;51.4;65.2;75.2
01008;AMBUTRIX;01;AIN;240100883;CC DE LA PLAINE DE L'AIN;92;65.2;75.2
01009;ANDERT-ET-CONDON;01;AIN;200040350;CC BUGEY SUD;34.2;65.2;52.4
[:~] $ tail /Users/alexandre/Downloads/txcouvglo2014.csv
97415;SAINT-PAUL;974;LA REUNION;249740101;CA TERRITOIRE DE LA COTE OUEST (TCO);33.2;25.8;29
97416;SAINT-PIERRE;974;LA REUNION;249740077;CA CIVIS (COMMUNAUTE INTERCOMMUNALE DES VILLES SOLIDAIRES);34.5;25.8;25.8
97417;SAINT-PHILIPPE;974;LA REUNION;249740085;CA DU SUD;16.9;25.8;18.4
97418;SAINTE-MARIE;974;LA REUNION;249740119;CA INTERCOMMUNALE DU NORD DE LA REUNION (CINOR);32;25.8;31.1
97419;SAINTE-ROSE;974;LA REUNION;249740093;CA INTERCOMMUNALE DE LA REUNION EST (CIREST);17.2;25.8;20
97420;SAINTE-SUZANNE;974;LA REUNION;249740119;CA INTERCOMMUNALE DU NORD DE LA REUNION (CINOR);28.1;25.8;31.1
97421;SALAZIE;974;LA REUNION;249740093;CA INTERCOMMUNALE DE LA REUNION EST (CIREST);17.7;25.8;20
97422;LE TAMPON;974;LA REUNION;249740085;CA DU SUD;20.3;25.8;18.4
97423;LES TROIS-BASSINS;974;LA REUNION;249740101;CA TERRITOIRE DE LA COTE OUEST (TCO);14.3;25.8;29
97424;CILAOS;974;LA REUNION;249740077;CA CIVIS (COMMUNAUTE INTERCOMMUNALE DES VILLES SOLIDAIRES);9.1;25.8;25.8
[:~] $ grep "2A" /Users/alexandre/Downloads/txcouvglo2014.csv
2A001;AFA;2A;CORSE DU SUD;242010056;CA DU PAYS AJACCIEN;32.6;35.8;27.6
2A004;AJACCIO;2A;CORSE DU SUD;242010056;CA DU PAYS AJACCIEN;29.5;35.8;27.6
2A006;ALATA;2A;CORSE DU SUD;242010056;CA DU PAYS AJACCIEN;20.8;35.8;27.6
```

```json
{
   "header":[
      "NUMCOM",
      "NOMCOM",
      "NUMDEP",
      "NOMDEP",
      "NUMEPCI",
      "NOMEPCI",
      "TXCOUVGLO_COM_2014",
      "TXCOUVGLO_DEP_2014",
      "TXCOUVGLO_EPCI_2014"
   ],
   "columns":{
      "NOMCOM":{
         "score":1.0,
         "format":"commune",
         "python_type":"string"
      },
      "NOMDEP":{
         "score":1.0,
         "format":"departement",
         "python_type":"string"
      },
      "NUMCOM":{
         "score":1.0,
         "format":"int",
         "python_type":"int"
      },
      "NUMDEP":{
         "score":1.0,
         "format":"int",
         "python_type":"int"
      },
      "NOMEPCI":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      },
      "NUMEPCI":{
         "score":1.0,
         "format":"siren",
         "python_type":"string"
      },
      "TXCOUVGLO_COM_2014":{
         "score":1.0,
         "format":"float",
         "python_type":"float"
      },
      "TXCOUVGLO_DEP_2014":{
         "score":1.0,
         "format":"float",
         "python_type":"float"
      },
      "TXCOUVGLO_EPCI_2014":{
         "score":1.0,
         "format":"float",
         "python_type":"float"
      }
   },
   "formats":{
      "int":[
         "NUMCOM",
         "NUMDEP"
      ],
      "float":[
         "TXCOUVGLO_COM_2014",
         "TXCOUVGLO_DEP_2014",
         "TXCOUVGLO_EPCI_2014"
      ],
      "siren":[
         "NUMEPCI"
      ],
      "string":[
         "NOMEPCI"
      ],
      "commune":[
         "NOMCOM"
      ],
      "departement":[
         "NOMDEP"
      ]
   },
   "encoding":"ISO-8859-1",
   "separator":";",
   "continuous":[
      "TXCOUVGLO_DEP_2014",
      "TXCOUVGLO_EPCI_2014"
   ],
   "categorical":[
      
   ],
   "total_lines":36636,
   "columns_fields":{
      "NOMCOM":{
         "score":1.0,
         "format":"commune",
         "python_type":"string"
      },
      "NOMDEP":{
         "score":1.0,
         "format":"departement",
         "python_type":"string"
      },
      "NUMCOM":{
         "score":1.0,
         "format":"code_commune_insee",
         "python_type":"string"
      },
      "NUMDEP":{
         "score":1.0,
         "format":"code_departement",
         "python_type":"string"
      },
      "NOMEPCI":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      },
      "NUMEPCI":{
         "score":1.0,
         "format":"siren",
         "python_type":"string"
      },
      "TXCOUVGLO_COM_2014":{
         "score":1.0,
         "format":"float",
         "python_type":"float"
      },
      "TXCOUVGLO_DEP_2014":{
         "score":0.9183673469387755,
         "format":"latitude_wgs",
         "python_type":"float"
      },
      "TXCOUVGLO_EPCI_2014":{
         "score":0.9387755102040817,
         "format":"longitude_wgs",
         "python_type":"float"
      }
   },
   "columns_labels":{
      "NOMCOM":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      },
      "NOMDEP":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      },
      "NUMCOM":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      },
      "NUMDEP":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      },
      "NOMEPCI":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      },
      "NUMEPCI":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      },
      "TXCOUVGLO_COM_2014":{
         "score":0.5,
         "format":"code_commune_insee",
         "python_type":"string"
      },
      "TXCOUVGLO_DEP_2014":{
         "score":0.5,
         "format":"code_departement",
         "python_type":"string"
      },
      "TXCOUVGLO_EPCI_2014":{
         "score":1.0,
         "format":"string",
         "python_type":"string"
      }
   },
   "header_row_idx":0,
   "heading_columns":0,
   "trailing_columns":0
}
```

### LINKED ISSUE #46: Add pandas profiling infos in csv detective report
Add in report for each columns : 
- min
- max
- mean
- std
- top 10 values with count for each
- nb distinct values

### PR DESCRIPTION
**Pull request by Pierlou**
Fix #40 
Fix #41 
Fix #42 
Fix #43 
Fix #44 
Fix #46

### CODE REVIEW

File: csv_detective/detect_fields/other/json/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import json
+
+PROPORTION = 1
+
+
+def _is(val):
+    '''Detects json'''
+    try:
+        loaded = json.loads(val)
+        if isinstance(loaded, dict) and not(any([geo in loaded for geo in ['coordinates', 'geometry']])):
```
Comment (abulte) [inline csv_detective/detect_fields/other/json/__init__.py:10]: What about lists?

**Review by abulte — COMMENTED** [review]
Nice 👏 ! Maybe add tests for new exposed types?

Also as we discussed, it would be nice to have a `date` type also.

File: csv_detective/detect_fields/other/json/__init__.py (hunk 1 above)
Comment (abulte) [reply csv_detective/detect_fields/other/json/__init__.py:10]: I might be wrong but I don't think anything else than an object (dict) or an array (list) can be expressed.

Comment (Pierlou) [reply csv_detective/detect_fields/other/json/__init__.py]: Changed ✅

File: tests/test_file.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from csv_detective import explore_csv
+import pytest
+
+def test_routine_on_file():
+    output_profile=True
```
Comment (geoffreyaldebert) [inline tests/test_file.py:5]:
```suggestion
```

File: tests/test_file.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    output = explore_csv.routine(csv_file_path="tests/a_test_file.csv",
+        num_rows=-1,
+        output_profile=output_profile,
```
Comment (geoffreyaldebert) [inline tests/test_file.py:8]:
```suggestion
        output_profile=True,
```

File: tests/test_file.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    assert output['columns']['NOMCOM']['format']=='commune'
+    assert output['columns']['NOMDEP']['format']=='departement'
+    assert output['columns']['NUMEPCI']['format']=='siren'
+    assert output['columns']['STRUCTURED_INFO']['python_type']=='json'
+    assert output['columns']['STRUCTURED_INFO']['format']=='json'
+    assert output['columns']['GEO_INFO']['python_type']=='json'
+    assert output['columns']['GEO_INFO']['format']=='json_geojson'
+    assert output['columns']['NUMEPCI']['format']=='siren'
+
+    if output_profile:
```
Comment (geoffreyaldebert) [inline tests/test_file.py:26]: Can we remove this ? we do not need it in this test

**Review by geoffreyaldebert — APPROVED** [review]

File: csv_detective/detection.py [hunk 5]
```diff
@@ -133,6 +135,51 @@ def parse_table(the_file, encoding, sep, num_rows, random_state=42):
...
+            if map_python_types.get(dict_cols_fields[c]['python_type'], str) in [float, int]:
+                profile[c].update(
+                    min=map_python_types.get(dict_cols_fields[c]['python_type'], str)(table[c].min()),
+                    max=map_python_types.get(dict_cols_fields[c]['python_type'], str)(table[c].max()),
+                    mean=map_python_types.get(dict_cols_fields[c]['python_type'], str)(table[c].mean()),
+                    std=map_python_types.get(dict_cols_fields[c]['python_type'], str)(table[c].std())
+                )
+            profile[c].update(
+                tops=[None if (isinstance(k, float) and np.isnan(k)) else k for k in list(table[c].value_counts(dropna=False).reset_index().iloc[:10].to_dict()['index'].values())],
+                nb_distinct=len(table[c].unique()),
```
[the remark this answers is not in the document]
Comment (Pierlou) [reply csv_detective/detection.py]: Je ne connaissais pas, je change !

File: tests/test_file.py (hunk 4 above)
Comment (Pierlou) [reply tests/test_file.py:26]: Remove the 2 asserts or the output_profile part?

Comment (geoffreyaldebert) [reply tests/test_file.py:26]: remove `if output_profile` and correct indentation of following lines

**Pull request merged**